src/main/java/uk/ac/ebi/embl/gff3tools/validation/builtin/AttributesRelationValidation.java

```diff
@@ -0,0 +1,213 @@
+            "Feature annotated with \"%s\" should not contain \"%s\". "
+                    + "Please move the \"%s\" attribute value into \"%s\" or add a comment for a curator.";
+
+    private static final String CIRCULAR_RNA_ATTRIBUTE_ERROR =
+            "Attribute circularRNA is not allowed in feature \"%s\",  only CDS,mRNA or tRNA feature can contain circularRNA qualifier.";
+
+    private static final Map<String, Set<String>> EXCLUSIVE_ATTRIBUTES =
+            Map.of("clone", Set.of("sub_clone", "clone_lib"));
+
+    public static final String MITOCHONDRION = "mitochondrion";
+    public static final String ENVIRONMENTAL_SAMPLE_VALUE_PATTERN = "^(uncultured).*";
+    public static final String PROVIRAL_VALUE_PATTERN = ".*endogenous retrovirus$";
+
+    private static final Map<String, Set<String>> MUTUALLY_EXCLUSIVE = Map.of(
+            GFF3Attributes.PROVIRAL, Set.of(GFF3Attributes.VIRION),
+            GFF3Attributes.REARRANGED, Set.of(GFF3Attributes.GERM_LINE),
+            GFF3Attributes.ENVIRONMENTAL_SAMPLE,
+                    Set.of(
+                            GFF3Attributes.STRAIN,
+                            GFF3Attributes.CULTIVAR,
+                            GFF3Attributes.VARIETY,
+                            GFF3Attributes.CULTURE_COLLECTION,
+                            GFF3Attributes.SPECIMEN_VOUCHER),
+            GFF3Attributes.PSEUDO, Set.of(GFF3Attributes.PSEUDOGENE, GFF3Attributes.PRODUCT),
+            GFF3Attributes.PSEUDOGENE, Set.of(GFF3Attributes.PRODUCT),
+            GFF3Attributes.CULTIVAR, Set.of(GFF3Attributes.VARIETY));
+
+    private static final Map<String, Set<String>> MUTUALLY_REQUIRED = Map.of(
+            GFF3Attributes.MAP, Set.of(GFF3Attributes.CHROMOSOME, GFF3Attributes.SEGMENT, GFF3Attributes.ORGANELLE),
+            GFF3Attributes.SUB_CLONE, Set.of(GFF3Attributes.CLONE),
+            GFF3Attributes.SUB_STRAIN, Set.of(GFF3Attributes.STRAIN),
+            GFF3Attributes.ENVIRONMENTAL_SAMPLE, Set.of(GFF3Attributes.ISOLATION_SOURCE),
+            GFF3Attributes.OLD_LOCUS_TAG, Set.of(GFF3Attributes.LOCUS_TAG),
+            GFF3Attributes.ORGANISM, Set.of(GFF3Attributes.ENVIRONMENTAL_SAMPLE));
+
+    // Map<DisallowedAttribute, Map<ConditionAttribute, Set<AttributeValues>>>
+    private static final Map<String, Map<String, Set<String>>> MUTUALLY_EXCLUSIVE_BY_VALUE = Map.of(
+            GFF3Attributes.ORGANELLE, Map.of(GFF3Attributes.GENE, Set.of("5.8S rRNA", "18S rRNA", "28S rRNA")),
+            GFF3Attributes.GERM_LINE, Map.of(GFF3Attributes.MOL_TYPE, Set.of("mRNA")),
+            GFF3Attributes.MACRO_NUCLEAR, Map.of(GFF3Attributes.ORGANELLE, Set.of(MITOCHONDRION)));
+
+    @ValidationMethod(rule = "GFF3_CIRCULAR_RNA_VALIDATION", type = ValidationType.FEATURE)
+    public void validateCircularRNAAttribute(GFF3Feature feature, int line) throws ValidationException {
+
+        String featureName = feature.getName();
+        if (feature.isAttributeExists(GFF3Attributes.CIRCULAR_RNA)) {
```

Comment:
The GFF3 feature will have the `Is_circular` attribute, which is equivalent to the `circular_RNA` on embl flat files. Before comparing attributes ensure you use the `qualifier-mapping.tsv`